cmd/sysl/codegen.go

```diff
@@ -128,13 +129,20 @@ func GenerateCode(
 	logger.Debugf("start: %s\n", codegenParams.Start)
 	logger.Debugf("basePath: %s\n", codegenParams.BasePath)
 
-	transformFs := syslutil.NewChrootFs(fs, codegenParams.RootTransform)
+	var transformFs afero.Fs
+	transformFs = syslutil.NewChrootFs(fs, codegenParams.RootTransform)
+	if mod.SyslModules {
+		transformFs = mod.NewFs(transformFs)
+	}
 	tfmParser := parse.NewParser()
 	tx, transformAppName, err := parse.LoadAndGetDefaultApp(codegenParams.Transform, transformFs, tfmParser)
 	if err != nil {
 		return nil, err
 	}
 
+	if mod.SyslModules {
+		fs = mod.NewFs(fs)
+	}
 	g, err := ebnfparser.ReadGrammar(fs, codegenParams.Grammar, codegenParams.Start)
```

Comment:
It is true to say that the grammar and transform are tightly coupled: transforms produce structures specified by grammars and parameterised by the SySL model. Grammars do proliferate less than transforms however; there could be multiple sets of transforms that all use the same singular grammar... so having the grammar read relative to the sysl executable may be correct.

Reply:
Ok, lets see how the current way works in practice when we reference transforms and grammars via modules instead of filesystem paths. We can change this later if we need to. Thanks Gregg.